### PR TITLE
Pivotal ID # 179867189: TSV Quotes Deserialization

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
https://www.pivotaltracker.com/story/show/179867189

- Fix regular expression to detect values inside quotes
- Add the corresponding test case

### CODE REVIEW

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/chunks/TsvChunkGenerator.kt [hunk 1]
```diff
@@ -71,7 +71,7 @@ internal class TsvChunkGenerator(private val parser: CSVFormat = createParser())
     private fun CSVRecord.asList(): List<String> = map { it }
 
     companion object {
-        private val SIMPLE_QUOTE_REGEX = "(\")([^\n|\t]*)(\")".toRegex()
+        private val SIMPLE_QUOTE_REGEX = "(\")([^\n|\t]*?)(\")".toRegex()
```
[the remark this answers is not in the document]
Comment (jhoanmanuelms) [reply commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/chunks/TsvChunkGenerator.kt:74]: The idea for this change is to catch every word surrounded by double quotes in the scenario where there're several quoted words in a single value which is the problem in this case. For example:

```
"10810064_Array2_output.txt.proc" OR "10810064_Array2_output.txt.proc" OR "10810062_Array2_output.txt.proc" OR "10810062_Array2_output.txt.proc" OR "10810065_Array1b_output.txt.proc" OR "10810065_Array1b_output.txt.proc"
```
The current regex only captures the very first and very last double quotes which are causing them to not be properly escaped. You can check the difference here:

Current regex: https://regex101.com/r/RCD9jQ/1

New regex: https://regex101.com/r/a8k9TQ/1

[the remark this answers is not in the document]
Comment (Juan-EBI) [reply commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/chunks/TsvChunkGenerator.kt:74]: I see your point thanks for explanation even having the same effect I will suggest to you to change expression to 

 `(\")([^\n|\t\"]*)(\")`
 
_An open quote , following by anything that is not a tab, line break or another quote. Following by a quote. Or change to_ 

`(\")([^\n|\t]?)(\")*`

_An open quote , following by anything that is not a tab, line break and a closing quote (n number of times)_ 


IMO both easier to undersdant , said that is clear that easier to undersdanrt is a subjective concept.

**Review by Juan-EBI — APPROVED** [review]
Please consider comment before merge

File: commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/chunks/TsvChunkGenerator.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhoanmanuelms) [reply commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/tsv/deserialization/chunks/TsvChunkGenerator.kt]: Done

**Pull request merged**